### PR TITLE
handling already tokenized and pos tagged data

### PR DESCRIPTION
**Pull request by pauloamed**
Still need to study how this modification will affect CN and JA languages.

### CODE REVIEW

File: tools/tokenizer/src/Tokenizer.java [hunk 1]
```diff
@@ -365,6 +365,32 @@ private static Analyzer getSuitableAnalyzer(String language, String mode) throws
         return arr;
     }
 
+    private static char PRETOK_DELIMITER = ' ';
```
Comment (remenberl) [inline tools/tokenizer/src/Tokenizer.java:368]: For better generality, we can make it as a FLAG string to allow any characters as delimiter. For example, --delimiter="\t, "

**Review by remenberl — COMMENTED** [review]
Thanks for the change. Overall looks good to me. Left a suggestion about making this functionality more flexible.

File: tools/tokenizer/src/Tokenizer.java (hunk 1 above)
Comment (pauloamed) [reply tools/tokenizer/src/Tokenizer.java:368]: Will do! Do we want this to imply `-pretok`? i.e. will both flags be needed?

Comment (remenberl) [reply tools/tokenizer/src/Tokenizer.java:368]: Using one flag also works if you prefer.

File: tools/tokenizer/src/Tokenizer.java [hunk 2]
```diff
@@ -836,6 +835,28 @@ private static void tokenizeText(String rawFileName, String targetFileName, Stri
         return;
     }
 
+    private static String PRETOK_DELIMITERS = "";
+    private static void setDelimiters(String rawDelimiters){
```
Comment (remenberl) [inline tools/tokenizer/src/Tokenizer.java:839]: please add space

Comment (remenberl) [reply tools/tokenizer/src/Tokenizer.java:839]: same for other places

**Review by remenberl — APPROVED** [review]

File: tools/tokenizer/src/Tokenizer.java (hunk 2 above)
Comment (pauloamed) [reply tools/tokenizer/src/Tokenizer.java:839]: will do!

**Review by remenberl — APPROVED** [review]

**Pull request merged**